From 8932d9dcb3cdd34dbdd59ed3d2513f84ddefc25a Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Tue, 11 Jun 2019 14:08:10 +0800
Subject: [PATCH 1/2] Fix APL_NUC boot issues without display connection

When the period is updated due to a config change,
cache the period and don't immediately update the
model until we actually observe updated vsyncs from the hardware.

But if never received the first resync sample, update the
mPeriod immediatel.

Change-Id: I3fd0b420033037f743a5458eed955525537a6b5f
Tracked-On: https://jira.devtools.intel.com/browse/OAM-81129
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 services/surfaceflinger/Scheduler/DispSync.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/surfaceflinger/Scheduler/DispSync.cpp b/services/surfaceflinger/Scheduler/DispSync.cpp
index cd6fa4194..1b701d443 100644
--- a/services/surfaceflinger/Scheduler/DispSync.cpp
+++ b/services/surfaceflinger/Scheduler/DispSync.cpp
@@ -647,6 +647,14 @@ void DispSync::setPeriod(nsecs_t period) {
         ATRACE_INT("DispSync:PendingPeriod", period);
     }
     mPendingPeriod = period;
+    // On this time, this is first get the period config, set it immediately update.
+    // Don't wait the hardware vsync.
+    if (mNumResyncSamples == 0) {
+        ALOGD("First resync sample doesn't received, update the mPeriod immediately");
+        mPeriod = period;
+        mPhase = 0;
+        mThread->updateModel(mPeriod, mPhase, mReferenceTime);
+    }
 }

 nsecs_t DispSync::getPeriod() {
--
2.21.0
